test/chol.jl

```diff
@@ -0,0 +1,25 @@
+@testset "Cholesky decomposition" begin
+    @testset "1×1" begin
+        m = @SMatrix [2.0]
+        (c,) = chol(m)
+        @test c === [2.0]
+    end
+
+    @testset "2×2" for i = 1:100
+        m_a = randn(2,2)
+        #non hermitian
+        @test_throws ArgumentError chol(SMatrix{2,2}(m_a))
+        m_a = m_a*m_a'
+        m = @SMatrix m_a
+        @test chol(m) ≈ chol(m_a)
+    end
+
+    @testset "3×3" for i = 1:100
+        m_a = randn(3,3)
```

Comment:
How about calling `srand()` somewhere so that these are fully reproducible?

Ideally we'd do large scale randomized testing (fuzzing, I guess) independently from the unit tests.